web_widget_mpld3_chart/static/src/js/web_widget_mpld3_chart.esm.js

```diff
@@ -1,31 +1,32 @@
 /** @odoo-module **/
 
-import basicFields from "web.basic_fields";
-import fieldRegistry from "web.field_registry";
-
-const Mpld3ChartWidget = basicFields.FieldChar.extend({
-    jsLibs: [
-        "/web_widget_mpld3_chart/static/src/lib/d3/d3.v5.js",
-        "/web_widget_mpld3_chart/static/src/lib/mpld3/mpld3.v0.5.9.js",
-    ],
-    _renderReadonly: function () {
+import {CharField} from "@web/views/fields/char/char_field";
+import {loadBundle} from "@web/core/assets";
+import {registry} from "@web/core/registry";
+const {onWillStart, markup} = owl;
+class Mpld3ChartWidget extends CharField {
```

Comment:
It would be much cleaner to use a JSON Field and use directly a component, as this fields are not editable

Review the Bokeh chart in order to see how it is managed:

https://github.com/OCA/web/blob/16.0/web_widget_bokeh_chart/readme/USAGE.rst#using-a-json-field

Reply:
@etobella Done!